cpp/src/parquet/column_writer.cc

```diff
@@ -993,7 +993,7 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
       WriteRepetitionLevels(num_values, rep_levels);
     } else {
       // Each value is exactly one row
-      rows_written_ += static_cast<int>(num_values);
+      rows_written_ += num_values;
```

Comment:
Also update this line?
https://github.com/apache/arrow/blob/563fceab6e0bff76c91c45624f9da5416f5fbe04/cpp/src/parquet/column_writer.cc#L1042